pandas/tests/frame/methods/test_drop.py

```diff
@@ -481,3 +481,13 @@ def test_drop_with_duplicate_columns2(self):
         df2 = df.take([2, 0, 1, 2, 1], axis=1)
         result = df2.drop("C", axis=1)
         tm.assert_frame_equal(result, expected)
+
+    def test_drop_pos_args_deprecation(self):
+        # https://github.com/pandas-dev/pandas/issues/41485
+        df = DataFrame({"a": [1, 2, 3]})
+        msg = (
+            r"Starting with Pandas version 2\.0 all arguments of drop except for the "
+            r"arguments 'self' and 'labels' will be keyword-only"
+        )
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            df.drop("a", 1)
```

Comment:
Can you add an assert that the result is still the same as `df.drop("a", axis=1)` (outside of the assert_produces_warning context)